core/src/test/java/cucumber/runtime/formatter/PrettyFormatterTest.java

```diff
@@ -381,6 +382,37 @@ public void should_mark_arguments_in_steps() throws Throwable {
                                           AnsiEscapes.GREEN + AnsiEscapes.INTENSITY_BOLD + "arg2"  + AnsiEscapes.RESET));
     }
 
+    // Reproduce issue #619: This test should now fail!
+    @Test(expected = StringIndexOutOfBoundsException.class)
+    public void nested_capture_groups_cannot_be_handled(){
```

Comment:
Ah. That means you have to write a test that would fail without your fix.

For example in your test you assert that it produces a certain output -the one you expected it to produce- but instead you get an exception. Then you fix the bug so the test passes.